### DISCUSSION TITLE
Issues running Pi Karaoke

### DISCUSSION THREAD

**Discussion by BentWrench** [vicwomg/pikaraoke#369] [General]
Hello All,

A couple of times in the past few months I have run Pi Karaoke and downloads won't work. The video starts to download and then disappears from the queue. When that happens I usually try updating yt-dlp via the Pi Karaoke link, but it always fails. I then have only been able to fix this by wiping the pikaraoke folder and re-installing. 

Is there a way to either fix the permissions so that the link works on the Pi Karaoke web page, or a separate command-line fix I can use? I tried updating yt-dlp but it must be in a non-standard location because that doesn't work either (using 'sudo yt-dlp -U' command).

One other issue I have had, after running sudo ./setup.sh and then launching pikaraoke ith ./pikaraoke.sh, Iget the error "PermissionError: [Errno 13] Permission denied: '/home/pi/pikaraoke/qrcode.png' and I fix that by running 'sudo ./pikaraoke.sh' one time and then it works.

Any help would be appreciated, this is getting frustrating. Thanks!

**Comment by vicwomg** [comment]
yt-dlp installs the to .venv directory as a dependency in order to isolate it from the rest of the filesystem. 

If you run ./setup.sh as sudo, this would cause all the issues you're describing. You should run it _without_ sudo: `./setup.sh` and also run pikaraoke.sh as non-sudo

**Reply by BentWrench** [reply]
OK, I'll try wiping it and reinstalling non-sudo, hopefully that works.
